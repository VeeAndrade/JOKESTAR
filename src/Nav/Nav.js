import React from 'react';
import { NavLink } from 'react-router-dom';
// import { connect } from 'react-redux'
import { useSelector } from 'react-redux'
import './Nav.scss';

const Nav = (props) => {
  const username = useSelector(state => state.username)
  return (
    <nav className='main-nav'>
      <h1 className='nav-title'><span role='img' aria-label='icon'>ðŸŒŸ</span>JOKE<span>STAR</span></h1>
        {!username ?
        <NavLink to='/login'><button className='nav-login-button'>Login</button></NavLink> : 
        <NavLink to='/'><button className='nav-login-button'>Logout</button></NavLink>}
    </nav>
  )
}



export default Nav;