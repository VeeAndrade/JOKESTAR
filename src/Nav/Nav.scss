// $faded-purple: rgba(85, 42, 102, 0.719);
$faded-purple: rgba(74, 21, 160, 0.452);
$bright-green: rgb(0, 255, 136);
.main-nav {
  width: 100%;
  height: 3rem;
  display: flex;
  justify-content: center;
  align-content: center;
  background: linear-gradient($faded-purple, $faded-purple);
}

.star-image {
  background: url('../Assets/star.png');
  width: 100px;
}

.nav-title {
  width: 92%;
  align-self: center;
  margin-left: 10px;
  font-size: 2em;
  color: #fff;
}

h1 span {
  color: $bright-green;
}

.nav-login-button {
  width: 8%;
  cursor: pointer;
  height: 80%;
  background-color: transparent;
  color: #fff;
  font-size: 1em;
  font-family: 'Questrial';
  font-weight: 400;
  align-self: center;
  margin-right: 8px;
  border-radius: 5px;
  border: solid 1px #fff;
  display: flex;
  justify-content: center;
  align-items: center;

  &:hover {
    background-color: #fff;
    color: rebeccapurple;
  }
}