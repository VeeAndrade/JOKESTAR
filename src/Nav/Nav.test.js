import React from 'react';
import { shallow } from 'enzyme';
import { Nav, mapStateToProps } from './Nav';

describe('Nav', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow( <Nav username={{username: ''}}/> )
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('mapStateToProps', () => {
  it('should return an array of favorites', () => {
    const mockState = {
      username: 'EvilKitty'
    }
    const expected = {
      username: 'EvilKitty'
    }
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected)
  });
});