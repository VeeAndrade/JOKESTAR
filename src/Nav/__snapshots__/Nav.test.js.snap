// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match the snapshot 1`] = `
<nav
  className="main-nav"
>
  <h1
    className="nav-title"
  >
    <span
      aria-label="icon"
      role="img"
    >
      â˜†
    </span>
    JOKE
    <span>
      STAR
    </span>
  </h1>
  <NavLink
    className="nav-login-button"
    to="/"
  >
    Logout
  </NavLink>
</nav>
`;
